sam/imageutils.py

```diff
@@ -1514,15 +1515,12 @@ def write_array_to_file(data_array, to_file, write_offset, hdfs_client=None):
     data = data_array.tobytes('F')
     if hdfs_client is None:
         seek_start = time()
-        with open(to_file, 'a+b') as f:
-            f.seek(write_offset, 0)
-            seek_number += 1
-            seek_time += time() - seek_start
-            write_start = time()
-            f.write(data)
-            f.flush()
-            os.fsync(f)
-            write_time += time() - write_start
+
+        fd=os.open(to_file, os.O_RDWR | os.O_APPEND)
```

Comment:
You removed the seek increment here. Are we considering seeks to only be the system call that increments the counter? Otherwise, seeks are still occurring. This, of course, is not important for the practical use of `sam`, but for benchmarking it would be.

Also, a perhaps misplaced comment: for your benchmarking did you have `os.fsync()` in clustered, multiple and naive but not in pwrites? that might explain at least a portion of the speedup.

Reply:
I did not use fsync in my benchmarking. For the count of seeks yes I thought about nb_seeks as the number of actual system calls to seek. But anyway I wanted to recheck all those counts and profiling in the week for my future tests.

Reply:
Also I plan to change the output value of the split and merge functions because the different functions return different values etc. I thought of a python dict but if you don't want it can still be a tuple.